src/django_components/commands/list.py

```diff
@@ -0,0 +1,221 @@
+import os
+from typing import Any, Dict, List, Optional, Type
+
+from django_components.component import all_components
+from django_components.util.command import CommandArg, ComponentCommand
+from django_components.util.misc import get_import_path, get_module_info
+
+
+# This descriptor generates the list of command arguments (e.g. `--all`), such that
+# their descriptions are dynamically generated based on the Command class.
+class ListArgumentsDescriptor:
+    # This will be called when accessing `ListCommand.arguments`
+    def __get__(self, obj: Optional["ListCommand"], cls: Type["ListCommand"]) -> List[CommandArg]:
+        command = obj or cls
+        all_cols = command.columns
+        default_cols = command.default_columns
+
+        all_cols_input = ",".join(all_cols)
+        all_cols_readable = ", ".join(all_cols)
+        default_cols_input = ",".join(default_cols)
+
+        return [
+            CommandArg(
+                ["--all"],
+                action="store_true",
+                # Format as e.g. `--columns name,full_name,path`
+                help=f"Show all columns. Same as `--columns {all_cols_input}`.",
+            ),
+            CommandArg(
+                ["--columns"],
+                help=(
+                    f"Comma-separated list of columns to show. Available columns: {all_cols_readable}. "
+                    f"Defaults to `--columns {default_cols_input}`."
+                ),
+            ),
+            CommandArg(
+                ["-s", "--simple"],
```

Comment:
One difference from what I described in https://github.com/django-components/django-components/issues/1024 is that I actually used `-s/--simple` instead of `-q/--quiet` to hide the table headers. Because I realized that `-q` is more commonly used when one wants to hide all outputs, which is not the case here.